tests/notifications/test-class-yoast-notification.php

```diff
@@ -0,0 +1,322 @@
+	}
+
+	/**
+	 * Test type custom value
+	 */
+	public function test_type_custom() {
+		$subject = new Yoast_Notification( 'message', array( 'type' => 'bla' ) );
+		$this->assertEquals( 'bla', $subject->get_type() );
+	}
+
+	/**
+	 * Test retrieval of dismissal key
+	 */
+	public function test_get_dismissal_key() {
+		$subject = new Yoast_Notification( 'message', array( 'dismissal_key' => 'dis' ) );
+		$this->assertEquals( 'dis', $subject->get_dismissal_key() );
+	}
+
+	/**
+	 * Test retrieval of dismissal key when not set
+	 */
+	public function test_get_dismissal_key_not_set() {
+		$subject = new Yoast_Notification( 'message', array( 'id' => 'id' ) );
+		$this->assertEquals( 'id', $subject->get_dismissal_key() );
+	}
+
+	/**
+	 * Non-persistent Notifications should always be displayed.
+	 */
+	public function test_not_persistent_always_show() {
+		$subject = new Yoast_Notification( 'message', array() );
+		$this->assertFalse( $subject->is_persistent() );
+		$this->assertTrue( $subject->display_for_current_user() );
+
+		$subject = new Yoast_Notification( 'message', array( 'capabilities' => array( 'foooooooo' ) ) );
+		$this->assertFalse( $subject->is_persistent() );
+		$this->assertTrue( $subject->display_for_current_user() );
+	}
+
+	/**
+	 * Test any without matches.
+	 */
+	public function test_match_any_pass() {
+
+		$me = wp_get_current_user();
+		$me->add_cap( 'bla' );
+		$me->remove_cap( 'foo' );
+
+		$subject = new Yoast_Notification(
+			'message',
+			array(
+				'id'               => 'id',
+				'capabilities'     => array(
+					'bla',
+					'foo',
+				),
+				'capability_check' => 'any',
+			)
+		);
+
+		$this->assertTrue( $subject->display_for_current_user() );
+
+		$me->remove_cap( 'bla' );
+	}
+
+	/**
+	 * Test any without matches.
+	 */
+	public function test_match_any_fail() {
+
+		$subject = new Yoast_Notification(
+			'message',
+			array(
+				'id'               => 'id',
+				'capabilities'     => array(
+					'bla',
+					'foo',
+				),
+				'capability_check' => 'any',
+			)
+		);
+
+		$this->assertFalse( $subject->display_for_current_user() );
+	}
+
+	/**
+	 * Test any without matches.
+	 */
+	public function test_match_all_pass() {
+
+		$me = wp_get_current_user();
+		$me->add_cap( 'bla' );
+		$me->add_cap( 'foo' );
+
+		$subject = new Yoast_Notification(
+			'message',
+			array(
+				'id'               => 'id',
+				'capabilities'     => array(
+					'bla',
+					'foo',
+				),
+				'capability_check' => 'all',
+			)
+		);
+
+		$this->assertTrue( $subject->display_for_current_user() );
+
+		$me->remove_cap( 'bla' );
+		$me->remove_cap( 'foo' );
+	}
+
+	/**
+	 * Test any without matches.
+	 */
+	public function test_match_all_fail() {
+
+		$subject = new Yoast_Notification(
+			'message',
+			array(
+				'id'               => 'id',
+				'capabilities'     => array(
+					'bla',
+					'foo',
+				),
+				'capability_check' => 'all',
+			)
+		);
+
+		$this->assertFalse( $subject->display_for_current_user() );
```

Comment:
A better check of `all` would be to have only one of the capabilities necessary. In this test `any` would fail as well.